bind.sh

```diff
@@ -13,92 +13,110 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#
-
-set -euo pipefail
 
-# Usage: bind.sh <launcher> [--cpus <spec>] [--gpus <spec>] [--mems <spec>] [--nics <spec>] <app> ...
+# Usage: bind.sh --launcher <launcher> [--cpus <spec>] [--gpus <spec>] [--mems <spec>] [--nics <spec>] -- <app> ...
 # <spec> specifies the resources to bind each node-local rank to, with ranks
 # separated by /, e.g. 0,1/2,3/4,5/6,7 for 4 ranks per node.
 
-# Detect node-local rank based on launcher
-IDX=none
-case "$1" in
-    mpirun) IDX="$OMPI_COMM_WORLD_LOCAL_RANK" ;;
-    jsrun) IDX="$OMPI_COMM_WORLD_LOCAL_RANK" ;;
-    srun) IDX="$SLURM_LOCALID" ;;
-    local) IDX=0 ;;
-    none) IDX="${SLURM_LOCALID:-${OMPI_COMM_WORLD_LOCAL_RANK:-${MV2_COMM_WORLD_LOCAL_RANK:-none}}}" ;;
+set -euo pipefail
+
+help() {
+  echo "Usage: bind.sh -l | --launcher <mpirun,srun,jrun,local> [ -c | --cpus ] [ -g | --gpus ] [ -m | --mems ] [ -n | --nics ] -- <app>"
+  exit 2
+}
+
+while :
+do
+  case "$1" in
+    --launcher) launcher="$2" ;;
+    --cpus) cpus="$2" ;;
+    --gpus) gpus="$2" ;;
+    --mems) mems="$2" ;;
+    --nics) nics="$2" ;;
+    --help) help ;;
+    --)
+      shift;
+      break
+      ;;
+    *)
+      echo "Unexpected option: $1"
+      help
+      ;;
+  esac
+  shift 2
+done
+
+if [ -z "$launcher" ]; then
+  echo "bind.sh: -l / --launcher is a required argument"
+fi
+
+case "$launcher" in
+  local) rank=0 ;;
+  mpirun) rank="$OMPI_COMM_WORLD_LOCAL_RANK" ;;
+  jsrun) rank="$OMPI_COMM_WORLD_LOCAL_RANK" ;;
+  srun) rank="$SLURM_LOCALID" ;;
```

Comment:
We still want to have the "catch-all" case for `launcher == none`. This is what would happen if legate is invoked with external launching, i.e. the user calls `mpirun -N 3 legate` instead of `legate --launcher mpirun`. In that case there will be one instance of `legate` invoked on each process, and we have no idea what launcher was used to launch us, so we just need to check every variable we know about.

Reply:
:+1:  OK we will need to remove the check above to make it required

Reply:
https://github.com/nv-legate/legate.core/pull/461/commits/cb05f269f6a078b470b319e448f3e558d261aa64